diff -u -r a/drivers/net/ethernet/mediatek/mtk_eth_soc.c b/drivers/net/ethernet/mediatek/mtk_eth_soc.c
--- a/drivers/net/ethernet/mediatek/mtk_eth_soc.c	2025-11-26 20:03:08.039000512 +0800
+++ b/drivers/net/ethernet/mediatek/mtk_eth_soc.c	2025-11-26 20:25:24.910001149 +0800
@@ -2652,6 +2652,7 @@
 		if (sent_ppd && ppd_dev){
 		skb->protocol = eth_hdr(skb)->h_proto;
 		skb->dev = ppd_dev;
+		skb->mark = 0x78681415;
 		}
 		skb_record_rx_queue(skb, 0);
 		napi_gro_receive(napi, skb);
diff -u -r a/net/bridge/br_input.c b/net/bridge/br_input.c
--- a/net/bridge/br_input.c	2025-11-26 20:03:07.524000511 +0800
+++ b/net/bridge/br_input.c	2025-11-26 20:27:01.545001195 +0800
@@ -137,7 +137,7 @@
 	nbp_switchdev_frame_mark(p, skb);
 
 	/* insert into forwarding database after filtering to avoid spoofing */
-	if (p->flags & BR_LEARNING)
+	if ((p->flags & BR_LEARNING) && (skb->mark != 0x78681415))
 		br_fdb_update(br, p, eth_hdr(skb)->h_source, vid, 0);
 
 	promisc = !!(br->dev->flags & IFF_PROMISC);
